### PR TITLE
Fix for Uri macro with `org` variable

### LINKED ISSUE #704: `Uri.uri` macro fails with symbol named org
Well, this is sad.

``` scala
scala> import org.http4s._

scala> Uri.uri("http://www.example.com/")
res0: org.http4s.Uri = http://www.example.com

scala> val bar = "bar"
bar: String = bar

scala> Uri.uri("http://www.example.com/") / bar
res1: org.http4s.Uri = http://www.example.com/bar

scala> val org = "foo"
org: String = foo

scala> Uri.uri("http://www.example.com/") / org
<console>:16: error: value http4s is not a member of String
       Uri.uri("http://www.example.com/") / org
              ^
```

### PR DESCRIPTION
**Pull request by samspills**
without the `_root_` in the uri macro, the added test fails to compile with `error: value http4s is not a member of String`

This issue was originally fixed in #741, but in the intervening years + refactors the `_root_` qualifier was dropped. I'm just re-adding it, and I'm adding it to all the literal macros just to be safe. 

fixes #704

### CODE REVIEW

**Comment by samspills** [comment]
Also realized that uri macro wasn't working if an implicit variable `org` was in scope 

```scala
scala> import org.http4s.implicits._
import org.http4s.implicits._

scala> implicit val org: String = "the-problem"
val org: String = the-problem

scala> uri"test"
       ^
       error: value http4s is not a member of String
```

Added a test for this behaviour as well and confirmed that it fails without [532c401](https://github.com/http4s/http4s/pull/7537/commits/532c40171c0a565d27276222a7a2f9c1c1055137) and passes with it.

File: tests/shared/src/test/scala/org/http4s/LiteralSyntaxMacrosSuite.scala [hunk 1]
```diff
@@ -18,10 +18,20 @@ package org.http4s
...
   test("'uri' macro works for valid input") {
     assertEquals(uri"a.b.c", Uri.fromString("a.b.c").toOption.get)
   }
+  test("'uri' macro works with variable named 'org'") {
+    val org = "blerf"
+    assertEquals(uri"org" / org, Uri.fromString("org/blerf").toOption.get)
+  }
+  test("'uri' macro works with implicit variable named 'org' in scope") {
+    @nowarn implicit val org = "the-problem"
+    assertEquals(uri"example", uri"example")
```
Comment (danicheg) [inline tests/shared/src/test/scala/org/http4s/LiteralSyntaxMacrosSuite.scala:33]: _nit_: maybe we need to check that `uri"example"` equals to `Uri.fromString("example").toOption.get`?
```suggestion
    assertEquals(uri"example", Uri.fromString("example").toOption.get)
```

**Review by danicheg — APPROVED** [review]
Thanks for fixing this!

File: tests/shared/src/test/scala/org/http4s/LiteralSyntaxMacrosSuite.scala [hunk 2]
```diff
@@ -18,10 +18,20 @@ package org.http4s
 
 import org.http4s.implicits._
 
+import scala.annotation.nowarn
+
 class LiteralSyntaxMacrosSuite extends Http4sSuite {
   test("'uri' macro works for valid input") {
     assertEquals(uri"a.b.c", Uri.fromString("a.b.c").toOption.get)
   }
+  test("'uri' macro works with variable named 'org'") {
```
Comment (hamnis) [inline tests/shared/src/test/scala/org/http4s/LiteralSyntaxMacrosSuite.scala:27]: would this now crash with `_root_` ?

Comment (samspills) [reply tests/shared/src/test/scala/org/http4s/LiteralSyntaxMacrosSuite.scala:27]: Yes, it starts failing with the error `value org is not a member of String`  (I added `implicit val _https_ = "blerf"`)

I think `_root_` as a variable name is a lot less common than `org` though?

**Review by hamnis — APPROVED** [review]

**Pull request merged**